packages/graphql-typescript-definitions/src/index.ts

```diff
@@ -278,3 +296,31 @@ function groupOperationsAndFragmentsByFile({
 
   return map;
 }
+
+function getFileNameFromPath(path: string) {
+  const fileName = basename(path);
+  const fileExtension = extname(path);
+  const extensionLength = fileExtension.length;
+
+  return fileName.slice(0, -extensionLength);
+}
+
+function checkFileNameMismatch(file: File) {
```

Comment:
I'm finding this a bit difficult to follow. Something more like `getFileNameFromPath` is easier to read.

```
const {operation} = file;

if (!operation) {
  return false;
}

const {filePath, operationName, operationType} = operation;
const fileName = getFileNameFromPath(operation.filePath);
const type = toCapitalized(operationType);

return fileName !== `${operationName}${type}`;
```

Reply:
😆 Don't know if that snippet even works. But I think you get what I mean

Reply:
Agreed 👍